eb_sqs/aws/sqs_queue_client.py

```diff
@@ -62,9 +62,21 @@ def add_message(self, queue_name, msg, delay):
         # type: (unicode, unicode, int) -> None
         try:
             queue = self._get_queue(queue_name)
-            queue.send_message(
-                MessageBody=msg,
-                DelaySeconds=delay
-            )
+            try:
+                queue.send_message(
+                    MessageBody=msg,
+                    DelaySeconds=delay
+                )
+            except ClientError as ex:
+                if ex.response.get('Error', {}).get('Code', None) == 'AWS.SimpleQueueService.NonExistentQueue':
+                    queue = self._get_queue(queue_name, use_cache=False)
+                    queue.send_message(
```

Comment:
Wouldn't it be cleaner to just call `add_message` again? Could be implemented with an internal `_add_message` taking a counter and use_cache flag to limit the recursion.

Reply:
the exception handling might be a bit confusing... e.g. the exception could be wrapped twice in `QueueClientException`